### PR TITLE
Alerting: Add ability to control scheduler tick interval via config

### PR DESCRIPTION
**Pull request by yuri-tceretian**
**What is this feature?**
This PR enables control over the scheduler's ticker interval via the configuration file.

The setting is controlled by a new experimental feature flag `configurableSchedulerTick`, which emphasizes the experimental nature of the change. It is not recommended to change it unless you know what you do. 

**Why do we need this feature?**
Currently, the interval is hard-coded and is 10s. This interval is also used to spread alert evaluation in order to lower the load on the data sources. However, with a lot of alert rules 10s is not enough (with just 100 rules the delay between evaluations is just 100ms). 
This is a way to remedy a situation when many rules make pressure on a data source and cause it to stop responding or rate-limiting the client. Currently, this is the only way until https://github.com/grafana/grafana/issues/53744 is implemented.

**Who is this feature for?**
Users have many rules that are not evaluated every 10s.

**Special notes for your reviewer:**
All possible side effects are described in the config file.

I tested the increasing of the default interval to 1m with rules that are evaluated every minute. The feature works as expected.

here is the graph that shows the evaluation rate of 500 rules with intervals of 1 minute.
image

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by JacobsonMT — APPROVED** [review]
LGTM

File: pkg/setting/setting_unified_alerting.go [hunk 1]
```diff
@@ -287,6 +287,27 @@ func (cfg *Cfg) ReadUnifiedAlertingSettings(iniFile *ini.File) error {
...
+	if cfg.IsFeatureToggleEnabled("configurableSchedulerTick") { // use literal to avoid cycle imports
+		if err != nil {
+			return fmt.Errorf("failed to parse setting 'scheduler_tick_interval' as duration: %w", err)
+		}
+		if baseInterval != SchedulerBaseInterval {
+			cfg.Logger.Warn("Scheduler tick interval is changed to non-default", "interval", uaCfg.BaseInterval, "default", SchedulerBaseInterval)
+		}
+		uaCfg.BaseInterval = baseInterval
+	} else if baseInterval != SchedulerBaseInterval {
+		cfg.Logger.Warn("Scheduler tick interval is changed to non-default but the feature flag is not enabled. Use default.", "interval", uaCfg.BaseInterval, "default", SchedulerBaseInterval)
```
Comment (grobinson-grafana) [inline pkg/setting/setting_unified_alerting.go:308]:
```suggestion
		cfg.Logger.Warn("Scheduler tick interval is changed to non-default but the feature flag is not enabled. Using default.", "interval", uaCfg.BaseInterval, "default", SchedulerBaseInterval)
```

File: pkg/setting/setting_unified_alerting.go [hunk 2]
```diff
@@ -287,6 +287,27 @@ func (cfg *Cfg) ReadUnifiedAlertingSettings(iniFile *ini.File) error {
...
+	// NOTE:
+	// 1. All alert rule intervals should be times of this interval. Otherwise, the rules will not be evaluated. It is not recommended to set it lower than 10s or odd numbers. Recommended: 10s, 30s, 1m
+	// 2. The increasing of the interval will affect how slow alert rule updates will reset the state, and therefore reset notification. Higher the interval - slower propagation of the changes.
+	baseInterval, err := gtime.ParseDuration(valueAsString(ua, "scheduler_tick_interval", SchedulerBaseInterval.String()))
+	if cfg.IsFeatureToggleEnabled("configurableSchedulerTick") { // use literal to avoid cycle imports
+		if err != nil {
+			return fmt.Errorf("failed to parse setting 'scheduler_tick_interval' as duration: %w", err)
+		}
+		if baseInterval != SchedulerBaseInterval {
+			cfg.Logger.Warn("Scheduler tick interval is changed to non-default", "interval", uaCfg.BaseInterval, "default", SchedulerBaseInterval)
```
Comment (grobinson-grafana) [inline pkg/setting/setting_unified_alerting.go:304]: When I run this with

```
scheduler_tick_interval = 30s
min_interval = 1m
```

I get the following log output:


```
WARN [07-26|16:40:52] Scheduler tick interval is changed to non-default logger=settings interval=10s default=10s
```

Shouldn't it be `30s` not `10s`?

**Review by grobinson-grafana — CHANGES_REQUESTED** [review]

**Review by grobinson-grafana — APPROVED** [review]
LGTM!

**Pull request merged**